torchvision/prototype/datasets/_builtin/pcam.py

```diff
@@ -0,0 +1,122 @@
+import io
+from torchvision.prototype import features
+from typing import Any, Callable, Dict, List, Optional, Tuple, Iterator
+
+import torch
+from torchdata.datapipes.iter import IterDataPipe, Mapper, Zipper
+from torchvision.prototype.datasets.utils import (
+    Dataset,
+    DatasetConfig,
+    DatasetInfo,
+    OnlineResource,
+    DatasetType,
+    GDriveResource,
+)
+from torchvision.prototype.datasets.utils._internal import (
+    hint_sharding,
+    hint_shuffling,
+)
+from torchvision.prototype.features import Label
+
+
+class H5Reader(IterDataPipe[Tuple[str, io.IOBase]]):
+    def __init__(
+        self,
+        datapipe: IterDataPipe[Tuple[str, io.IOBase]],
+        key: Optional[str] = None,  # Note: this key thing might be very specific to the PCAM dataset
+    ) -> None:
+        self.datapipe = datapipe
+        self.key = key
+
+    def __iter__(self) -> Iterator[Tuple[str, io.IOBase]]:
+        import h5py  # noqa
+        for _, handle in self.datapipe:
+            with h5py.File(handle) as data:
```

Comment:
Note: using the file_path (`_`) works just as well here. @pmeier are there any subtleties that would make us choose one over the other?

Reply:
1. The file is already open so there is no need to open it again. I'm guessing this works on Linux and Mac, but could maybe bite us on Windows.
2. For this dataset the files are present on disk directly. If they live inside an archive, the path is just an indicator and not a real path, like `/foo/bar.zip/baz.h5`. In this case we can only use the handle.

For these reasons I would always go with the handle directly unless there is a reason not to.